pkg/collect/run.go

```diff
@@ -140,38 +140,55 @@ func runPod(ctx context.Context, client *kubernetes.Clientset, runCollector *tro
 		serviceAccountName = runCollector.ServiceAccountName
 	}
 
-	pod := corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      runCollector.CollectorName,
-			Namespace: namespace,
-			Labels:    podLabels,
-		},
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "v1",
-			Kind:       "Pod",
-		},
-		Spec: corev1.PodSpec{
-			RestartPolicy:      corev1.RestartPolicyNever,
-			ServiceAccountName: serviceAccountName,
-			Containers: []corev1.Container{
-				{
-					Image:           runCollector.Image,
-					ImagePullPolicy: pullPolicy,
-					Name:            "collector",
-					Command:         runCollector.Command,
-					Args:            runCollector.Args,
+	pod := corev1.Pod{}
+	if runCollector.PodSpec.Containers != nil {
+		pod = corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      runCollector.CollectorName,
+				Namespace: namespace,
+				Labels:    podLabels,
+			},
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Pod",
+			},
+			Spec: runCollector.PodSpec,
+		}
+	} else {
+		pod = corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      runCollector.CollectorName,
+				Namespace: namespace,
+				Labels:    podLabels,
+			},
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Pod",
+			},
+			Spec: corev1.PodSpec{
+				RestartPolicy:      corev1.RestartPolicyNever,
+				ServiceAccountName: serviceAccountName,
+				Containers: []corev1.Container{
+					{
+						Image:           runCollector.Image,
+						ImagePullPolicy: pullPolicy,
+						Name:            "collector",
+						Command:         runCollector.Command,
+						Args:            runCollector.Args,
+					},
 				},
 			},
-		},
-	}
+		}
 
-	if runCollector.ImagePullSecret != nil && runCollector.ImagePullSecret.Data != nil {
-		secretName, err := createSecret(ctx, client, pod.Namespace, runCollector.ImagePullSecret)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to create secret")
+		if runCollector.ImagePullSecret != nil && runCollector.ImagePullSecret.Data != nil {
```

Comment:
is this useful when there is a podspec?

Reply:
when `podSpec` isn't used you'd still be able to use this feature, but planned to make it clear in the docs that wiht `podSpec` you'd also have to specify an existing secret name in the